### PR TITLE
added more documentation

### PR DESCRIPTION
**Pull request by zacharysyoung**
My goals were:

- an example for every type of scanner
- pkgsite-style links (hopefully not going "link happy")
- uniform language for similar scanners and options

### CODE REVIEW

File: csv/scanner.go [hunk 1]
```diff
@@ -1,11 +1,25 @@
...
+// Advance the scanners with the Scan method and check errors with
+// the Error method (unlike fields and fixedwidth, which use Err).
 package csv
 
 import (
 	"encoding/csv"
 	"io"
 )
 
-// Scanner wraps a csv.Reader via an API similar to that of bufio.Scanner.
```
Comment (zacharysyoung) [inline csv/scanner.go:8]: The note about bufio.Scanner got moved to the top-level scanners.go [doc string](https://github.com/smartystreets/scanners/pull/7/files#diff-d79e5d1d5bdbf80413e72607249dd87ba03228ca459f796ef7e385fb366f5914R4).

File: csv/column_scanner_test.go [hunk 2]
```diff
@@ -69,6 +69,20 @@ func (this *ColumnScannerFixture) TestColumnNotFound_Panic() {
 	this.So(func() { this.scanner.Column("nope") }, should.Panic)
 }
 
+// TestDuplicateColumnNames confirms that duplicated/repeated
```
Comment (zacharysyoung) [inline csv/column_scanner_test.go:72]: Slightly out of scope, but as I was documenting the Column method asked myself what would happen if the header had duplicate field names.

File: README.md [hunk 3]
```diff
@@ -4,3 +4,5 @@
 [![Code Coverage](https://codecov.io/gh/smartystreets/scanners/branch/master/graph/badge.svg)](https://codecov.io/gh/smartystreets/scanners)
 [![Go Report Card](https://goreportcard.com/badge/github.com/smartystreets/scanners)](https://goreportcard.com/report/github.com/smartystreets/scanners)
 [![GoDoc](https://godoc.org/github.com/smartystreets/scanners?status.svg)](http://godoc.org/github.com/smartystreets/scanners)
+
+Explore [the package](https://pkg.go.dev/github.com/smartystreets/scanners) on pkg.go.dev.
```
Comment (mdwhatcott) [inline README.md:8]: The link on line 6, while somewhat outdated, redirects to pkg.go.dev, so this extra link is somewhat superfluous.

Comment (zacharysyoung) [reply README.md]: Updated badge link; removed superfluous link.

**Comment by mdwhatcott** [comment]
Thanks @zacharysyoung!

**Pull request merged**